pkg/sql/types/minimum_type_version.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package types
+
+import "github.com/cockroachdb/cockroach/pkg/clusterversion"
+
+// regroleTypeVersion is an alias for MarkerDataKeysRegistry to avoid
+// having to backport a new version.
+const regroleTypeVersion = clusterversion.MarkerDataKeysRegistry
+
+// minimumTypeUsageVersions defines the minimum version needed for a new
+// data type.
+// Note: please do not remove this map or IsTypeSupportedInVersion even
+// if the map becomes empty temporarily.
+var minimumTypeUsageVersions = map[*T]clusterversion.Key{
+	RegRole: regroleTypeVersion,
+}
+
+// IsTypeSupportedInVersion returns whether a given type is supported in the given version.
```

Comment:
Would it simplify the code if you returned an error rather than bool from this helper? 

I can see it both ways.

Reply:
All the errors are the same so it would simplify the code in that sense but I prefer returning a bool over an error for these type of functions, I find that sometimes I come back to these type of functions that return errors and wish they returned a bool instead for flexibility.

Reply:
I feel that. I think there's a middle ground which is to create a helper to generate the error:

```go
package sql

func checkTypeIsSupported(ctx context.Context, settings *cluster.Settings, typ *types.T) error {
    version := settings.Version.ActiveVersionOrEmpty(ctx)
    if supported := types.IsTypeSupportedInVersion(version, typ); !supported {
        return pgerror.Newf(
            pgcode.FeatureNotSupported,
            "type %s is not supported until version upgrade is finalized",
            typ.SQLString(),
	)
    }
} 
```

Reply:
Nice suggestion, done.